src/Neo/SmartContract/ApplicationEngine.Runtime.cs

```diff
@@ -398,7 +398,10 @@ protected internal void SendNotification(UInt160 hash, string eventName, Array s
             Notify?.Invoke(this, notification);
             notifications ??= new List<NotifyEventArgs>();
             notifications.Add(notification);
-            CurrentContext.GetState<ExecutionContextState>().NotificationCount++;
+            if (CurrentContext != null)
```

Comment:
This is required to test without context. And this change will not change existing system as otherwise the system should have crashed.

Reply:
I don't think we need to change the implementation just for tests. Just mock the executing state in the test itself, see for example the https://github.com/neo-project/neo/blob/fdc90f988e08b9f308068c76fa5fb93b6ec4435b/tests/Neo.UnitTests/SmartContract/UT_ApplicationEngine.cs#L151-L153

Reply:
Previously it doesn't fail?

Reply:
if current context is null, then there is no tx executed.

Reply:
> I don't think we need to change the implementation just for tests. Just mock the executing state in the test itself, see for example the following test: https://github.com/neo-project/neo/blob/fdc90f988e08b9f308068c76fa5fb93b6ec4435b/tests/Neo.UnitTests/SmartContract/UT_ApplicationEngine.cs#L151-L153

null warning will be solved sooner or later, mock also work. i will check that.